src/main/java/org/wildfly/security/http/impl/NonceManager.java

```diff
@@ -94,7 +94,7 @@ String generateNonce(byte[] salt) {
 
             ByteBuffer byteBuffer = ByteBuffer.allocate(PREFIX_LENGTH + messageDigest.getDigestLength());
             byteBuffer.putInt(nonceCounter.incrementAndGet());
-            byteBuffer.putLong(System.currentTimeMillis());
+            byteBuffer.putLong(System.nanoTime());
```

Comment:
One other glitch here: nanoTime doesn't have an absolute start or end, so it is only meaningful when subtracted from another nanoTime from the same JVM instance.  If this nonce goes to live outside of the JVM, it could be a problem.